arcgis-ios-sdk-samples/Edit data/Edit and sync features/EditAndSyncFeaturesViewController.swift

```diff
@@ -0,0 +1,280 @@
+            self.generateJob = nil
+            self.barButtonItem.isEnabled = false
+            self.instructionsLabel.text = "Tap on a feature"
+            self.mapView.touchDelegate = self
+        }
+    }
+    
+    func geodatabaseDidSync() {
+        self.presentAlert(title: "Geodatabase sync sucessful")
+        self.barButtonItem.isEnabled = false
+        self.instructionsLabel.text = "Tap on a feature"
+    }
+    
+    func generateGeodatabase() {
+        // Hide the unnecessary items.
+        barButtonItem.isEnabled = false
+        extentView.isHidden = true
+        
+        // Get the area outlined by the extent view.
+        areaOfInterest = self.extentViewFrameToEnvelope()
+        
+        geodatabaseSyncTask.defaultGenerateGeodatabaseParameters(withExtent: areaOfInterest) { [weak self] (params: AGSGenerateGeodatabaseParameters?, error: Error?) in
+            if let params = params,
+                let self = self {
+                // Don't include attachments to minimize the geodatabase size.
+                params.returnAttachments = false
+                
+                // Create a temporary file for the geodatabase.
+                let dateFormatter = ISO8601DateFormatter()
+                
+                let documentDirectoryURL = FileManager.default.urls(for: .documentDirectory, in: .userDomainMask).first!
+                let downloadFileURL = documentDirectoryURL
+                    .appendingPathComponent(dateFormatter.string(from: Date()))
+                    .appendingPathExtension("geodatabase")
+               
+                // Request a job to generate the geodatabase.
+                let generateGeodatabaseJob = self.geodatabaseSyncTask.generateJob(with: params, downloadFileURL: downloadFileURL)
+                self.generateJob = generateGeodatabaseJob
+                generateGeodatabaseJob.start(
+                    statusHandler: { (status: AGSJobStatus) in
+                        SVProgressHUD.show(withStatus: status.statusString()) //Show job status.
+                    },
+                    completion: { [weak self] (_, error: Error?) in
+                        SVProgressHUD.dismiss()
+                        
+                        if let error = error {
+                            self?.presentAlert(error: error)
+                        } else {
+                            // Load the geodatabase when the job is done.
+                            self?.geodatabase = generateGeodatabaseJob.result
+                            self?.geodatabase.load { [weak self] (error: Error?) in
+                                self?.geodatabaseDidLoad()
+                            }
+                        }
+                    }
+                )
+            } else {
+                self!.presentAlert(title: "Could not generate default parameters: \(error!)")
+            }
+        }
+    }
+    
+    func syncGeodatabase() {
+        barButtonItem.isEnabled = false
```

Comment:
I just tried this method but I don't think it will work because the geodatabase does not yet exist upon the first instance of generating the geodatabase.

Reply:
why would the sync method be called before generating geodatabase?

Reply:
The Sync button doesn't become enabled until after a feature has been moved, so checking for whether the geodatabase has local edits would be superfluous.